composer.json

```diff
@@ -13,26 +13,38 @@
     "prefer-stable": true,
     "require": {
         "php": ">=8.1",
-        "doctrine/doctrine-bundle": "^2.10.0",
         "makinacorpus/query-builder": "^1.6.1",
+        "psr/log": "^3.0",
         "symfony/config": "^6.0|^7.0",
-        "symfony/console": "^6.0|^7.0",
-        "symfony/dependency-injection": "^6.0|^7.0",
         "symfony/filesystem": "^6.0|^7.0",
         "symfony/finder": "^6.0|^7.0",
         "symfony/options-resolver": "^6.0|^7.0",
-        "symfony/password-hasher": "^6.0|^7.0",
         "symfony/process": "^6.0|^7.0",
         "symfony/yaml": "^6.0|^7.0"
     },
     "require-dev": {
+        "doctrine/doctrine-bundle": "^2.10.0",
         "doctrine/orm": "^2.15|^3.0",
         "friendsofphp/php-cs-fixer": "^3.34",
         "phpstan/phpstan": "^1.10",
         "phpunit/phpunit": "^10.4",
+        "symfony/console": "^6.0|^7.0",
+        "symfony/dependency-injection": "^6.0|^7.0",
         "symfony/framework-bundle": "^6.0|^7.0",
+        "symfony/password-hasher": "^6.0|^7.0",
         "symfony/validator": "^6.3|^7.0"
     },
+    "suggest": {
+        "composer/pcre": "For PHAR compilation",
+        "doctrine/doctrine-bundle": "For autoconfiguration in Symfony project context",
+        "seld/phar-utils": "For PHAR compilation",
+        "symfony/password-hasher": "In order to use the password hash anonymizer"
+    },
+    "conflict": {
+        "composer/pcre": "<3.1|>=4.0",
+        "seld/phar-utils": "<1.2|>=2.0",
+        "symfony/password-hasher": "<6.0|>=8.0"
```

Comment:
May be we should add some restriction with the doctrine/doctrine-bundle here ? Or on doctrine/orm ?

Reply:
`makinacorpus/query-builder` already handles `doctrine/dbal` and that's the whole point of delegating connection handling to it. Although, we still have one single class where we use it directly: in the database session manager. I think it's OK to add restrictions, but let's be very open with it, such as `^3|^4` for DBAL and `^2|^3` for ORM.

Reply:
Done.